### ISSUE TITLE
Wordpress/WooCommerce wrapper/client

### ISSUE THREAD

**Issue by codeagencybe** [colloqi/pisignage-server#81]
Hello

Does anybody know of any existing solution/plugin for CMS like Wordpress and WooCommerce? 
My goal is to fetch content (like blogs or products, promotions, ...) from a Wordpress/WooCommerce website and load it dynamically into a presentation. 
This way, one of our clients can automatically update the Signage based on changes in their website. 

If anybody already did something similar, mind sharing your idea and tips? Or if there is a commercial addon available somewhere, I'm happy to hear about that too. 

Thanks!

**Comment by hitesh83** [comment]
If my understanding of your use case is correct, I would be doing this in below two ways

Option -1
Ask your client to give you permanent URL, which he can update time to time.
And Pisignage will display same url , you can make content for x second, so after every x second page will refresh and you will get new content.
Same can be done via JS

Option -2
Host a html file on your WP server , this html file will redirect to content of your clients page, you can either use redirector plugin in WordPress or manually edit. 

By both ways your client can change content without anyone visiting Pisignage dashboard.

**Comment by codeagencybe** [comment]
@hitesh83 

No, I'm talking about creating a dynamic template in PiSignage which fetch data over Wordpress/WooCommerce API. 

_I'm coming from SignageLive competitor service,_ and they have this service/feature more or less "out of the box". 
I'm now moving over to PiSignage as it is more budget friendly and open source to have more control what we can do with the content and features. 

With this I mean, SignageLive has an addon/module that allows creating a dynamic template eg a promotion box. 
Next, you can connect an API service to it and it replaces fields dynamically from the template. 
So you can design eg a promotion box with a product image, regular price, sale price, from/end date for the promotion valid, a description etc... 
All those data are fetched realtime by API from WooCommerce. 
You create a playlist and add this template to playlist and it slides/rotates between the number of records it finds by API. 
This way, you can schedule e.g. day deals, week deals, .... 

My client does not has to do anything at all. Not desiging or changing in PiSignage, no special pages/links to create in WooCommerce. 
They just change/set the product data in their shop which they already have to do for the shop itself. 
The signage just syncs in copy with that information.

**Comment by hitesh83** [comment]
As of today no such plugin available for Pisignage, but you can write a small code to do this which will start with player from boot and run in background and serve a local html , which Pisignage can display as per your playlist.
We did similar thing for displaying data from a API.
